module/modmanager/manager.go

```diff
@@ -739,7 +749,7 @@ func (mgr *Manager) newOnUnexpectedExitHandler(mod *module) func(exitCode int) b
 		// Finally, handle orphaned resources.
 		var orphanedResourceNames []resource.Name
 		for name, res := range mod.resources {
-			if _, err := mgr.addResource(mgr.restartCtx, res.conf, res.deps); err != nil {
+			if _, err := mgr.AddResource(mgr.restartCtx, res.conf, res.deps); err != nil {
```

Comment:
Yep I like it! Not really sure why I used the non-locking version here. So this change gets rid of the data race you were seeing that also existed on `main`?